src/test/kotlin/graphics/scenery/tests/examples/basic/ArcballExample.kt

```diff
@@ -19,60 +19,56 @@ class ArcballExample : SceneryBase("ArcballExample") {
     override fun init() {
         renderer = hub.add(Renderer.createRenderer(hub, applicationName, scene, 1024, 1024))
 
-        val cam: Camera = DetachedHeadCamera()
-        with(cam) {
+        val cam: Camera = DetachedHeadCamera {
             spatial {
                 position = Vector3f(0.0f, 0.0f, 2.5f)
             }
             perspectiveCamera(70.0f, windowWidth, windowHeight)
 
             targeted = true
             target = Vector3f(0.0f, 0.0f, 0.0f)
-
-            scene.addChild(this)
         }
+        scene += cam
 
-        val camlight = PointLight(3.0f)
-        camlight.intensity = 5.0f
-        cam.addChild(camlight)
-
-        val box = Box(Vector3f(1.0f, 1.0f, 1.0f))
+        val camlight = PointLight(3.0f) {
+            intensity = 5.0f
+        }
+        cam += camlight
```

Comment:
This is a neat trick. But I find it odd to override `plusAssign` but not `plus`. The implication, which is not true for these types, is that `cam + camlight` would be a new camera object with the light added, and only because the `+=` was used does it get assigned back into the same `cam` variable. Is this a pattern Kotlin users often use for mutators, to override only the `+=` but not `+`?

Reply:
Yep

Reply:
Huh, I guess it's convenient, so I'll swallow my feelings of mathematical discomfort :laughing:

Reply:
We might say this is one of the example on the border line, some people will assume this to be an operator abuse, others don't

But that's where I like to give the choice :)